spec/javascripts/govuk_publishing_components/load-analytics.spec.js

```diff
@@ -0,0 +1,323 @@
+  'identity.company-information.service.gov.uk',
+  'identity-sandbox.company-information.service.gov.uk',
+  'import-products-animals-food-feed.service.gov.uk',
+  'lastingpowerofattorney.service.gov.uk',
+  'live.email-dvla.service.gov.uk',
+  'live.dvla-web-chat.service.gov.uk',
+  'loststolenpassport.service.gov.uk',
+  'makeaplea.service.gov.uk',
+  'managefleetvehicles.service.gov.uk',
+  'manage-apprenticeships.service.gov.uk',
+  'manage-fish-exports.service.gov.uk',
+  'manage-quota.fluorinated-gas.service.gov.uk',
+  'manage-water-abstraction-impoundment-licence.service.gov.uk',
+  'match.redundancy-payments.service.gov.uk',
+  'mot-testing.service.gov.uk',
+  'nominate-uk-honour.service.gov.uk',
+  'notice.redundancy-payments.service.gov.uk',
+  'passport.service.gov.uk',
+  'paydvlafine.service.gov.uk',
+  'payments.service.gov.uk',
+  'publish-payment-practices.service.gov.uk',
+  'queens-awards-enterprise.service.gov.uk',
+  'recruit.manage-apprenticeships.service.gov.uk',
+  'register.fluorinated-gas.service.gov.uk',
+  'register-trailer.service.gov.uk',
+  'register.ozone-depleting-substances.service.gov.uk',
+  'registertovote.service.gov.uk',
+  'register-vehicle.service.gov.uk',
+  'registers.service.gov.uk',
+  'reminders.mot-testing.service.gov.uk',
+  'renewable-heat-calculator.service.gov.uk',
+  'reply-jury-summons.service.gov.uk',
+  'report-director-conduct.service.gov.uk',
+  'report.fluorinated-gas.service.gov.uk',
+  'report.ozone-depleting-substances.service.gov.uk',
+  'right-to-rent.homeoffice.gov.uk',
+  'right-to-work.service.gov.uk',
+  'ruralpayments.service.gov.uk',
+  'schools-financial-benchmarking.service.gov.uk',
+  'secured.studentfinanceni.co.uk',
+  'secured.studentfinancewales.co.uk',
+  'selfservice.payments.service.gov.uk',
+  'send-money-to-prisoner.service.gov.uk',
+  'signin.service.gov.uk',
+  'sorn.service.gov.uk',
+  'staff.helpwithcourtfees.service.gov.uk',
+  'student-finance.service.gov.uk',
+  'tax.service.gov.uk',
+  'teacherservices.education.gov.uk',
+  'teaching-vacancies.service.gov.uk',
+  'to-visit-or-stay-in-the-uk.homeoffice.gov.uk',
+  'trade-tariff.service.gov.uk',
+  'tribunal-response.employmenttribunals.service.gov.uk',
+  'ukri.org',
+  'update-student-loan-employment-details.service.gov.uk',
+  'vehicle-operator-licensing.service.gov.uk',
+  'vehicleenquiry.service.gov.uk',
+  'viewdrivingrecord.service.gov.uk',
+  'view-and-prove-your-rights.homeoffice.gov.uk',
+  'view-immigration-status.service.gov.uk',
+  'visa-address-update.service.gov.uk',
+  'visas-immigration.service.gov.uk',
+  'your-defra-account.defra.gov.uk'
+]
+
+var expected
+var savedUaVars
+var savedGa4Vars
+
+describe('Analytics loading', function () {
+  beforeAll(function () {
+    savedUaVars = window.GOVUK.extendObject(window.GOVUK.analyticsVars)
+    window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {}
+    savedGa4Vars = window.GOVUK.extendObject(window.GOVUK.analyticsGa4.vars)
```

Comment:
I'm wondering if there's an easier way of saving and switching around the stored variables here. Presumably some of this effort is to avoid impacting other tests? Could the spec helper by default call the analytics loading with a dud environment, to prevent it loading everywhere else? Could the DOMContentLoaded be wrapped in some logic relating to the environment instead of containing it? Could `window.GOVUK.loadAnalytics` be called only in these tests, and then the relevant window variables be deleted afterwards?

Reply:
Yep this code is to prevent it affecting the other tests - the tests were going haywire and the only way I could fix it was to preserve the previous analytics vars as the other tests seem to be dependent on them being structured a certain way.

I'm not sure what you mean with the suggestions - I don't think there's a way to determine if you're running jasmine tests which would stop `window.GOVUK.loadAnalytics` running as Jasmine is running on `localhost` so it would load the development UA/GTM variables onto `window.GOVUK`. 

If you called `window.GOVUK.loadAnalytics` and it set the variables, and then you delete them afterwards, the other tests will fail (mainly the UA ones) because they need something set on `window.GOVUK.analyticsVars` so if I deleted them they will fail. That's why I set them to their previous value.

Does that make sense? You might have to provide the solution you're proposing here as I'm not sure what the alternative would be. Thanks

Reply:
It's been three weeks since I wrote that comment and I'm not sure what I was saying either! Seems good to me.